From b6da1c2c16cb5e2ca4fb8a39032a04d258f0f719 Mon Sep 17 00:00:00 2001
From: Matan Breizman <mbreizma@redhat.com>
Date: Thu, 26 Jun 2025 11:04:46 +0300
Subject: [PATCH] crimson/common/smp_helpers: fix missing include
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With 550b2b8ade7e10078ab9610442d039b65cce6a2b introducing coroutines in
smp_helpers header, GCC 15 FTBFS with:
```
FAILED: src/crimson/CMakeFiles/crimson.dir/net/ProtocolV2.cc.o
In file included from /home/zhscn/project/ceph/src/crimson/net/io_handler.h:11,
                 from /home/zhscn/project/ceph/src/crimson/net/ProtocolV2.h:9,
                 from /home/zhscn/project/ceph/src/crimson/net/ProtocolV2.cc:4:
/home/zhscn/project/ceph/src/crimson/common/smp_helpers.h: In function ‘decltype (seastar::futurize_invoke(f)) crimson::invoke_on_all_seq(F)’:
/home/zhscn/project/ceph/src/crimson/common/smp_helpers.h:64:7: error: coroutines require a traits template; cannot find ‘std::coroutine_traits’ [-Wtemplate-body]
   64 |       co_await crimson::submit_to(core, [&f] { return seastar::futurize_invoke(f);});
      |       ^~~~~~~~
/home/zhscn/project/ceph/src/crimson/common/smp_helpers.h:64:7: note: perhaps ‘#include <coroutine>’ is missing
```

Signed-off-by: Matan Breizman <mbreizma@redhat.com>
---
 src/crimson/common/smp_helpers.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/crimson/common/smp_helpers.h b/src/crimson/common/smp_helpers.h
index 80d56b086ba06..d9e2b61e82ab7 100644
--- a/src/crimson/common/smp_helpers.h
+++ b/src/crimson/common/smp_helpers.h
@@ -15,6 +15,7 @@
 #include "common/likely.h"
 #include "crimson/common/errorator.h"
 #include "crimson/common/utility.h"
+#include "crimson/common/coroutine.h"
 
 namespace crimson {
 
